packages/yoga/src/Datepicker/web/Datepicker.jsx

```diff
@@ -0,0 +1,323 @@
+    z-index: 1;
+    width: ${components.input.width}px;
+    left: 0;
+    border-radius: ${radii.regular}px;
+    margin-top: ${spacing.xxsmall}px;
+  `}
+`;
+
+const CalendarIcon = styled(({ _disabled, inputFilled, ...props }) =>
+  !inputFilled ? (
+    <Booking width={20} height={20} {...props} />
+  ) : (
+    <BookingFilled width={20} height={20} {...props} />
+  ),
+)`
+  ${({
+    disabled,
+    theme: {
+      yoga: { colors, spacing },
+    },
+  }) => `
+    fill: ${disabled ? colors.text.disabled : colors.text.primary};
+    margin-right: ${spacing.xxsmall}px;
+  `}
+`;
+
+const ErrorWrapper = styled(Text.Small)`
+  ${({
+    theme: {
+      yoga: { colors, components },
+    },
+  }) => `
+    margin-top: ${components.input.helper.margin.top}px;
+    font-size: ${components.input.helper.font.size}px;
+    color: ${colors.feedback.attention[1]};
+    position: absolute;
+  `}
+`;
+
+export const toUTC = date => {
+  return new Date(
+    date.getUTCFullYear(),
+    date.getUTCMonth(),
+    date.getUTCDate(),
+    date.getUTCHours(),
+    date.getUTCMinutes(),
+    date.getUTCSeconds(),
+  );
+};
+
+function Datepicker({
+  fullWidth,
+  type,
+  placeholder,
+  startDate,
+  endDate,
+  onSelectSingle,
+  disabled,
+  onSelectRange,
+  disablePastDates,
+  disableFutureDates,
+  error,
+  onOpen,
+}) {
+  const [open, setOpen] = useState();
+  const [startD, setStartDate] = useState(startDate);
+  const [endD, setEndDate] = useState(endDate);
+  const ref = useRef(null);
+  const [inputFilled, setInputFilled] = useState(false);
+
+  const triggerOnOpen = useCallback(() => {
+    if (onOpen) onOpen(open === 'true');
+  }, [open]);
+
+  const onBlur = () => {
+    setOpen(false.toString());
+    triggerOnOpen();
+  };
+
+  useEffect(() => {
+    if (type === 'single' && onSelectSingle) {
+      onSelectSingle(startD);
+      setOpen(false.toString());
+    } else if (type === 'range' && onSelectRange) {
+      onSelectRange(startD, endD);
+    }
+    setInputFilled(!!startD);
+  }, [startD, endD]);
+
+  const onDateSingleSelect = startLocal => {
+    setStartDate(startLocal);
+  };
+
+  useEffect(() => {
+    if (open === 'true') ref.current.focus();
+  }, [open]);
+
+  const onDateRangeSelect = selectedDate => {
+    if (!endD) {
+      if (!startD) {
+        setStartDate(selectedDate);
+      } else if (selectedDate < startD) {
+        setStartDate(selectedDate);
+        setEndDate(undefined);
+      } else setEndDate(selectedDate);
+    } else {
+      setStartDate(selectedDate);
+      setEndDate(undefined);
+    }
+  };
+
+  const renderInput = () => {
+    if (!startD && !endD) {
+      return (
+        <InputPlaceholder disabled={disabled}>
+          {placeholder ?? `Select Date`}
+        </InputPlaceholder>
+      );
+    }
+
+    const dateFormat = 'MMM D, YYYY';
+
+    return (
+      startD && (
+        <Input disabled={disabled} data-testid="datepicker-input">
```

Comment:
> Using data-testid attributes do not resemble how your software is used and should be avoided if possible. ([Testing Library docs](https://testing-library.com/docs/queries/bytestid/))


Can you check how you can handle your tests by avoiding using these test ids? 
In this case, to find the Input you can use the `screen.getByRole('textbox')`

I hardly suggest you do the same with other testIds you're declaring within your component.

Reply:
@tomaskeong

Reply:
Thank you for the review, I completely agree with your comment @frgiovanna. I've removed every testid except for the two arrow icons that allow the user to go the previous/next month. I couldn't find a way of dealing with those.